concrete/composer.json

```diff
@@ -61,7 +61,7 @@
     "concrete5/zend-queue": "0.9.*",
     "zendframework/zend-mail": "2.7.*",
     "zendframework/zend-cache": "2.7.*",
-    "zendframework/zend-http": "2.6.*",
+    "zendframework/zend-http": "2.8.1",
```

Comment:
That's possible, but there are no reported vulnerabilities in any versions higher than 2.8.1 according to roave/security-advisories.

The problem is that 2.8.1 requires PHP 5.6+ but concrete5 v8.x still supports PHP 5.5.9, so this will have to wait until support for that version is dropped.

Reply:
Yes, we nedd to wait for concrete5 9.

About the version constraint:
- `"zendframework/zend-http": "2.8.1"` instructs composer to install version `2.8.1` (and only it).
- `"zendframework/zend-http": "^2.8.1"` instructs composer to install version `2.8.1`, or any versions that's compatible (in [semver terms](https://getcomposer.org/doc/articles/versions.md#caret-version-range-)) with `2.8.1`

I usually prefer the second approach, because if some other library requires for example 2.9, it won't be possible to use it (concrete5 requires 2.8.1 and the other library 2.9).

Reply:
It'd be nice if we could drop support for PHP `5.5` in a minor version and go with PHP `5.6`. Maybe it's worth discussing given it could effect security.

Reply:
I've read that someone considers requiring a newer PHP version is not a breaking change.
For me it's a huge breaking change.

Furthermore, I may not see the big picture, but I don't think that [the security issue fixed zf/http version 2.8.1](https://github.com/zendframework/zend-http/commit/5234f4a9e8137b731ab95d6a17879d4eb8fb9e39#diff-575f824f107b9f3f4ed6f4d73830ca8c) is so relevant: if attackers can forge the request headers, they can still forge the `IIS_WasUrlRewritten` / `UNENCODED_URL` headers...

Reply:
The thing is that PHP 5.5 is no longer secure, so security releases for projects that see upgrading *a minor version* of PHP as not being a BC break will all be using PHP 5.6 at this point and we cannot upgrade - even if the security fix is extremely pressing.

I would definitely agree that going from PHP 5.2 to PHP 5.3 minimum is a huge BC break given the distance between those releases, however, I wouldn't say the same about going from PHP 5.5 to 5.6 because it's not as big of a leap. As long as composer has PHP 5.6 minimum set, composer based concrete5 will not break (since it just won't update until you have the proper PHP minimum version), and we can do the same with in-dashboard upgrades.